### ISSUE TITLE
Ground primitive artifact on iPad

### ISSUE THREAD

**Issue by OmarShehata** [AnalyticalGraphicsInc/cesium#8288]
[Sandcastle link](https://sandcastle.cesium.com/index.html#c=hVfbbttGEP0VVi+lYIXe+yV1g7pO2wRo0SIx+hIFBS2tJKI0KZArp27gf+9ZksvoUjl6kbh7ZnbmzJnl6CFvkofCfXJN8n1SuU/JjWuL3X32Z7eWfrvoHm/qyudF5Zpvp9/Nqx6ftQtXuWxd1ncuW7qt39y61l+vgWv9rWsa/IDPVV62DkZzD0Mc5ovStc4fntZ/8de3/Wb6Odk15cu4+7au3rm23jULl62a+v66BebtMpVa0GnydBLStinuC188uDbLl8t0ODHAjoCL/N41ebYqH2/r9PO8SvBZIomiyn1RV2MAN3nj8SuveHf+a7dunGvTF1pkhBhlDKdcM8HsLBEkU8YapS0T2hIpZgkVkkxnvfe6KRyo7L0PJ4bPxuXLolqPJ/6W+03m63dYzas25ZJHD+GzLfxicw77got9bFOXYJJkpF96mldPAxMoxqLM27ZYFYsuoj/q8nFdVy1q82GIbT5pm7+qej55mQg1i2uLum6WbVicTzoSKCGaUs4JFzMwoAlnnFLBOKGzniUrmbBGMBIBioI0KgSPAG6kVFzSCBAWvHHUeARIpgQzTA8ASZWSnJARADwB6cJGgNWCKy7YADBAwCHV0YNV0koiOR0BlEiLyFgEMCuUVjhyABCpGWfaRg+MMk0E1aMHIgIZygz71EIXxshxn2pGpcbBEUCRBqFGRQAjjCIHrSIArGpOxAhQOBVk8sEDM4qGIOQYo+JwYLlQEWAMFdx88WAgTsnA5ADQ1HCEjJgGIhlVBsRHB5JawYTidva1Ys8no0by0hd+t3S9SuAxI3w+gfpmyam2KLHnxUUM54ZLlJLGsgTCjCV8TBmlh3g47wGCKt5Vwoy0wwY52iHioCwjhLRRO9pAb9CKjB4YF1pYZUfSzsZwNmfwdDZhSO6ZfIMkDSo8qAyShA4MlXIUSXiEVIYKSa1RIlR8rCBBuRk3UUUKzafD19htRIZ0FIv72lIliTXjProN3RNbRTGlWUDZEQCZaE6ZiQBoGppS6osHSsE5jQAebCDFsRWMCM0keYxRoo1QFTYWTWkqDGVs6Gc8UlwhRI1Fs3CA5uNjFpC6omMIBt0HUpiKDiAMUMu0mn2N6GeELDMlzgsZ99kzhdUczpmJvStwI6HNmIkZGWEEAmIs6hQXnEGvGzsC0GjoFkGjkKEFFtQ86tSGLNGtEYBWYEzx8RI0VhNcQZFVdLbGTcnYl+63JASlY5DSWkqVHYO0TEkoCZdrDJJKJYgkI61n0zxLq6WZ7q+Hj93LaVU3SRreUNv+nZTUqzMvq+lQBLzph6V029QYW9qbwH86OMj6akxn0WMWAwhvw6fuzF21CJ73XfVWswTg6bz6HMaY7r1Zl3UYmOJ8EB6zdz+9zj4VfnNdbjd5SjLZvWf9c6PJ/gh0mF3EpXtDwtrV9843j28xYOUVUkxe7g9Rvxxt75vumx9aDZlG42OjJPL1psCo1Cw2/28+7qbH5snzQ9Q15sTHN65Yb3w7sD09drE/zsRxKRggknSbN637uaxzn4YaJRcJI6d4949vUOrlm2hHDhFPxya5901xt/Mdx59Pc+oF8PJAAMf8X0cfXb4dJO3sTuNLknZTfzpk9j1WzrpMkY87Iuppz+1BQoeNc/u4dXuRn+xlP/5++2YYFqenzXHQXHe7ogxz683QJn1d9jvFu/stGuUQmLXbsvBpuA0mQ5MAusLYXb5HgtUaFrgMJt0WroLuJiiwSL7D11XnNCtdtfYbLFxcTAOuO3Lui1W67+ib3tO034S3vb2LsJfMhnNO9sIpH4qPFwgTuIv+8YKOC32ygy2i6H4Fso5iGCJNXiVkenpQiOEDvO2t4YCPQ1CN87umStxDXu677MoymU2uWv9Yuld9tX4o7rd148NfpzTLLkO8ZQ4FX97tFn87ny3aNthdXUajq2XxkBTL7/GqOvyXB1Y6zWBntSvL98W/uKZfXV0Cf2CGtgtF/f3BNWX+GCAb+urXfjHLsqtLPJ5a+bou7/Jmz+N/). This works fine on desktop, and mostly fine on iPad:

![correct](https://user-images.githubusercontent.com/1711126/66861099-61e5f180-ef5c-11e9-9934-4013c476c482.png)

Until you zoom in a bit on iPad, you can see a seam very similar to the one in https://github.com/AnalyticalGraphicsInc/cesium/pull/8205 that moves with the camera that will cut out pieces of the polygon:

![incorrect](https://user-images.githubusercontent.com/1711126/66861104-63afb500-ef5c-11e9-8bad-1c4ece75c711.png)

I believe it's related to https://github.com/AnalyticalGraphicsInc/cesium/pull/8205 because I can see the same seam across the 3D Tileset, but using that branch the seam is fixed on the 3D Tileset but remains on the ground polygons.

**Comment by lilleyse** [comment]
How many frustums are being drawn in that view. It might not be #8205 if it's just one.

@likangning93 might have a better idea.

**Comment by lilleyse** [comment]
Oh except this is iPad which may not support log depth, so there are probably multiple frustums.

**Comment by OmarShehata** [comment]
Yes, I suspect there's multiple frustums because iPad doesn't support log depth. Although I can't reproduce the issue on desktop by turning off log depth. I updated the Sandcastle to turn off log depth and include the inspector. You can see it's using 2 frustums:

![two_frustums](https://user-images.githubusercontent.com/1711126/66933718-79c67f80-f007-11e9-8bb3-8e0c54231005.png)

And at that camera distance that "shreds" the polygon, if you show the frustum geometry and rotate around, you can see it's when the frustum plane intersects that building:

![frustum_intersect](https://user-images.githubusercontent.com/1711126/66933724-7b904300-f007-11e9-910f-aa4909ad5f1b.png)
